### ISSUE TITLE
Homerow randomly loses focus of my browser (Arc browser)

### ISSUE THREAD

**Issue by michaelessiet** [nchudleigh/homerow#124]
**Describe the bug**

Whenever I open or cmd+tab to my browser I see that homerow works perfectly fine and as intended, however, after a few minutes of browsing homerow doesn't highlight any of the links inside the browser window. It only highlights the browser's UI features like the menu bar, settings, back, forward, reload and window manipulation buttons.

Notice that none of the github buttons or links are accessible in the screenshot below

**Screenshots / video**
![Screenshot 2024-08-14 at 23 26 15](https://github.com/user-attachments/assets/c9afbb1b-1232-482b-96a6-b6af94d4a417)

**Steps to reproduce the bug**

1. Make sure that homerow is running
2. Open Arc browser
3. Surf the web for 15-30mins
4. Try navigating using homerow

**Comment by greg-wolff** [comment]
same here, except, arc web content has stopped working with homerow for me a while ago. i can select tabs from the sidebar, but web content is unselectable.

**Comment by saitoi** [comment]
Same here, it only goes back to working whenever I change focus and then change back to Arc.
After about 5 minutes using Arc, the Homerow app seems to not be able to trace the UI elements of Arc (only the help) located on the far left.

**Comment by michaelessiet** [comment]
It's really weird. I also noticed that if you have a window manager like aerospace running homerow doesn't work with spotify... which is kinda weird

**Comment by godbout** [comment]
Arc automatically deactivates its AX tree after a while of inactivity.

**Comment by michaelessiet** [comment]
@godbout you're right. I'm starting to notice this with even shortcat, the shortcuts and overlays work but after not interacting with the Arc window directly the browser shortcuts like cmd+L or ctrl+tab don't work... I honestly don't want to switch browsers again 🤦‍♂️

**Comment by godbout** [comment]
@michaelessiet [pinged the Arc devs about it](https://github.com/godbout/kindaVim.docs/issues/229#issuecomment-2024663159) almost half a year ago but no dice. i do apps that use the macOS AX Framework and deal with third party apps so i'm used to contacting developers to ask them to handle the AX properly and my chance of success so far after 4 years has been 0%. hehe.

**Comment by michaelessiet** [comment]
That really sucks. Hopefully they get around to addressing it sooner than later. BTW thanks for the follow on X. Were you able to find a work around for this in KV?

**Comment by godbout** [comment]
> Were you able to find a work around for this in KV?

depending on the issue i sometimes just ping the devs and move on, digging into the issue again only if they answer back. i don't use Arc myself so no, i didn't check more. but that should affect Wooshy too. although i know for sure about a good bunch of users that use Arc and never reported. hmm. will bump this up in the list of priorities and see. maybe some sort of timer might help, but i'd like to avoid that. will see.

**Comment by almeidainf** [comment]
+1
Just started using Homerow today and noticed this too. I haven't used it enough to identify a pattern, but sometimes Homerow labels the tabs and other UI elements in Arc, but nothing within the web page.

I will add to this if I learn more.
Great app! Excited to reduce my mouse use.

**Comment by nick-skriabin** [comment]
same experience

>  I also noticed that if you have a window manager like aerospace running homerow doesn't work with spotify... which is kinda weird

oh, wow, didn't even think about it. so when it "suddenly" stopped working is when I switched from yabai to Aerospace. huuuuh..

**Comment by godbout** [comment]
> Were you able to find a work around for this in KV?

ended up creating a new custom engine for Arc. didn't do before because i thought it was overkill, but there seems to be a new trend in apps that use Electron for UI and they're doing an even shittier job than before so i end up creating a bunch of custom engines for them 🤣️🤣️🤣️ so Arc support is gonna get improved in the next release of kV/Ws/Sl. can keep you updated if needed
